apps/base-docs/tutorials/docs/2_coinbase-smart-wallet.md

```diff
@@ -0,0 +1,694 @@
+author: briandoyle81
+keywords: [Solidity, ERC-721, token, NFT, wagmi, viem, smart wallet]
+tags: ['nft', 'smart wallet']
+difficulty: intermediate
+hide_table_of_contents: false
+displayed_sidebar: null
+---
+
+The [Coinbase Smart Wallet] is a new application of account abstraction that makes it easy to give the exact same onboarding to both native crypto users who have their own EOA wallet, and new users who would benefit from an easier onboarding to crypto. For the former, the wallet works almost exactly the way they are used to. For the latter, [passkeys] are used to create a wallet and securely save the keys.
+
+Most importantly, this wallet will work for the user across all apps using the Smart Wallet, across many different chains! Think of it as an onchain version of the social logins that most regular users now expect in their apps.
+
+---
+
+## Objectives
+
+By the end of this tutorial you should be able to:
+
+- Connect users to your onchain app using the Coinbase Smart Wallet
+- Integrate the Smart Wallet with wagmi
+- Provide an easy path for your users to buy crypto and put it in their wallet
+- Compare and contract the Smart Wallet with other forms of account abstraction
+- Take advantage of the [Base Gasless Campaign] to onboard your first users with free gas
+
+---
+
+## Prerequisites
+
+### ERC-721 Tokens
+
+This tutorial assumes that you are able to write, test, and deploy your own ERC-721 tokens using the Solidity programming language. If you need to learn that first, check out our content in [Base Camp] or the sections specific to [ERC-721 Tokens]!
+
+### Vercel
+
+You'll need to be comfortable deploying your app to [Vercel], or using another solution on your own. Check out our tutorial on [deploying with Vercel] if you need a refresher!
+
+### Onchain Apps
+
+The tutorial assumes you're comfortable with the basics of deploying an app and connecting it to a smart contract. If you're still learning this part, check out our tutorials in [Base Camp] for [Building an Onchain App].
+
+---
+
+## Building the Contract
+
+Start by setting up an [OpenZeppelin ERC-721] contract. You'll need to set up a `mintTo` function that accepts the address that should receive the NFT.
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+contract RandomColorNFT is ERC721 {
+    uint public counter;
+
+    constructor() ERC721("RandomColorNFT", "RCNFT") {
+    }
+
+    function mintTo(address _to) public {
+        counter++;
+        _safeMint(_to, counter);
+    }
+}
+```
+
+:::caution
+
+You need to be careful how you handle `msg.sender` when you are using a smart wallet with account abstraction. Unlike a normal transaction with a traditional wallet, `msg.sender` is **not** the address of the user's wallet address. It is instead their smart contract address.
```

Comment:
Do we have an explainer article on this concept? maybe reference openzeppelins Context's _msgSender()